### PR TITLE
[LLVMCPU] Renaming distribution and vectorization related variables.

### PR DESCRIPTION
**Pull request by hanhanW**
It revisits the naming and comments and replaces flow/workgroup to preferred term (i.e., distribution/vectorization).

- Replace flow* and workgroup* related terms for distribution.
- Replace wrong workgroup* terms for vectorization.

### CODE REVIEW

**Comment by hanhanW** [comment]
@MaheshRavishankar @dcaballe here is the draft PR about renaming. We can land it after landing Diego's multi-tiling work. (I think that's easier for addressing conflicts. :p)

File: compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp [hunk 1]
```diff
@@ -67,11 +68,9 @@ static llvm::cl::list<int> mmt4dVectorSizes(
     "iree-codegen-llvm-mmt4d-vector-size",
     llvm::cl::desc("linalg.mmt4d vector tile size"), llvm::cl::ZeroOrMore);
 
-static llvm::cl::opt<int> defaultWorkgroupTileSize(
-    "iree-codegen-llvm-generic-ops-workgroup-size",
-    llvm::cl::desc(
-        "linalg.generic and linalg.indexed_generic workgroup tile size"),
-    llvm::cl::init(64));
+static llvm::cl::opt<int> defaultDistTileSize(
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp]: Nit: `defaultDistTilSizes` -> `defaultDistributionTileSizes` (just being consistent with above).

**Review by MaheshRavishankar — APPROVED** [review]

File: compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp (hunk 1 above)
Comment (dcaballe) [reply compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp]: I'm a bit confused... Aren't distribution and flow level tile sizes the ones that were equivalent, same for workgroup and parallel/reduction tile sizes? Why are we replacing workgroup with distribution?

I think the replacement should be s/flow/distribution and s/workgroup/vector/. Then we will have vector parallel, vector reduction, cache parallel, cache reduction... At least I have followed that convention in multi-level tiling. WDYT?

File: compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp [hunk 2]
```diff
@@ -373,8 +372,7 @@ static unsigned getReferenceTypeLengthInBytes(func::FuncOp entryPointFn) {
   return referenceTypeLengthInBytes;
 }
 
-/// Returns the default tile sizes to use for the loops that are distributed at
-/// Flow level.
+/// Returns the default tile sizes to use for the loops that are distributed.
 static SmallVector<int64_t> getDefaultDistributedLoopTileSizes(
```
Comment (dcaballe) [inline compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp]: -> getDefaultDistributionTileSizes?

File: compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp (hunk 1 above)
Comment (hanhanW) [reply compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp]: Some workgroup term was mis-used in this file, and I replaced them with the correct one -- which is distribution. There were two flags about tile sizes. One is `*workgroup*` and the other is `*vector*`. I think the former one should be replaced with `distribution`, does it make sense?

**Pull request merged**